src/libs/logging.ts

```diff
@@ -0,0 +1,163 @@
+
+      await cockpitLogsDB.setItem(fileName, this.currentCockpitLog)
+    }, interval)
+  }
+
+  /**
+   * Stop the current logging operation
+   */
+  stopLogging(): void {
+    if (!this.logging()) {
+      Swal.fire({ title: 'Error', text: 'No log is being generated.', icon: 'error', timer: 3000 })
+      return
+    }
+
+    clearInterval(this.currentLoggerInterval)
+  }
+
+  /**
+   * Wether the logger is currently logging or not
+   * @returns {boolean}
+   */
+  logging(): boolean {
+    return this.currentLoggerInterval !== undefined
+  }
+
+  /**
+   * Get desired part of a log based on timestamp
+   * @param {CockpitStandardLog} completeLog The log from which the slice should be taken from
+   * @param {Date} initialTime The timestamp from which the log should be started from
+   * @param {Date} finalTime The timestamp in which the log should be terminated
+   * @returns {CockpitStandardLog} The actual log
+   */
+  getSlice(completeLog: CockpitStandardLog, initialTime: Date, finalTime: Date): CockpitStandardLog {
+    return completeLog
+      .filter((logPoint) => logPoint.epoch > initialTime.getTime() && logPoint.epoch < finalTime.getTime())
+      .map((logPoint) => ({ ...logPoint, ...{ seconds: differenceInSeconds(new Date(logPoint.epoch), initialTime) } }))
+  }
+}
+
+export const assFromClog = (log: CockpitStandardLog, videoWidth: number, videoHeight: number): string => {
+  let assFile = `[Script Info]
+Title: Cockpit Subtitle Telemetry file
+ScriptType: v4.00+
+WrapStyle: 0
+ScaledBorderAndShadow: yes
+YCbCr Matrix: TV.601
+PlayResX: ${videoWidth}
+PlayResY: ${videoHeight}
+
+[V4+ Styles]
+Format: Name, Fontname, Fontsize, PrimaryColour, SecondaryColour, OutlineColour, BackColour, Bold, Italic, Underline, StrikeOut, ScaleX, ScaleY, Spacing, Angle, BorderStyle, Outline, Shadow, Alignment, MarginL, MarginR, MarginV, Encoding
+Style: Default,Arial,48,&H00FFFFFF,&H000000FF,&H00000000,&H00000000,0,0,0,0,100,100,0,0,1,2,2,1,10,10,10,1
+
+[Events]
+Format: Layer, Start, End, Style, Name, MarginL, MarginR, MarginV, Effect, Text`
+
+  log.forEach((logPoint) => {
+    const roll = Number(logPoint.data['roll']).toFixed(2)
+    const pitch = Number(logPoint.data['pitch']).toFixed(2)
+    const heading = Number(logPoint.data['heading']).toFixed(2)
+    const depth = Number(logPoint.data['depth']).toFixed(2)
+    const mode = logPoint.data['mode']
+    const batteryVoltage = Number(logPoint.data['batteryVoltage']).toFixed(2)
+    const batteryCurrent = Number(logPoint.data['batteryCurrent']).toFixed(2)
+    const gpsVisibleSatellites = logPoint.data['gpsVisibleSatellites']
+    const gpsFixType = logPoint.data['gpsFixType']
+    const latitude = Number(logPoint.data['latitude']).toFixed(6)
+    const longitude = Number(logPoint.data['longitude']).toFixed(6)
+    const secondsStart = logPoint.seconds.toFixed(0)
+    const secondsFinish = (logPoint.seconds + 1).toFixed(0)
```

Comment:
Can we include units for these somehow? The indicators already include units, so that seems like it should be doable, hopefully :-)

Reply:
[relevant comment](https://github.com/bluerobotics/cockpit/issues/450#issuecomment-1823806753)

Reply:
I will try the discussed approach. If it fits, I will change the PR.

Reply:
On the new approach, each widget register which variables it uses, and this information is used for automatic variable selection (if no user custom selection is made).